cql3/statements/strongly_consistent_select_statement.cc

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ *
+ * Modified by ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
+ */
+
+
+#include "cql3/statements/strongly_consistent_select_statement.hh"
+
+#include <seastar/core/future.hh>
+
+#include "cql3/restrictions/statement_restrictions.hh"
+#include "cql3/query_processor.hh"
+#include "raft/broadcast_tables/experimental/lang.hh"
+
+namespace cql3 {
+
+namespace statements {
+
+static
+bytes get_key(const cql3::expr::expression& partition_key_restrictions) {
+    const auto* conjunction = cql3::expr::as_if<cql3::expr::conjunction>(&partition_key_restrictions);
+
+    if (!conjunction || conjunction->children.size() != 1) {
+        throw raft::broadcast_tables::unsupported_operation_error(fmt::format(
+            "partition key restriction: {}", partition_key_restrictions));
+    }
+
+    const auto* key_restriction = cql3::expr::as_if<cql3::expr::binary_operator>(&conjunction->children[0]);
+
+    if (!key_restriction) {
+        throw raft::broadcast_tables::unsupported_operation_error(fmt::format("partition key restriction: {}", *conjunction));
+    }
+
+    const auto* column = cql3::expr::as_if<cql3::expr::column_value>(&key_restriction->lhs);
+    const auto* value = cql3::expr::as_if<cql3::expr::constant>(&key_restriction->rhs);
+
+    if (!column || column->col->kind != column_kind::partition_key ||
+        !value || key_restriction->op != cql3::expr::oper_t::EQ) {
+        throw raft::broadcast_tables::unsupported_operation_error(fmt::format("key restriction: {}", *key_restriction));
```

Comment:
This should be caught in the prepare stage, yes?

So this should be on_internal_error().

Reply:
Ah, I was confused since you're fetching the value here. Values are fetched during the execution stage since not all values are constant.